src/arch/riscv/tlb.cc

```diff
@@ -422,6 +428,14 @@ TLB::translate(const RequestPtr &req, ThreadContext *tc,
         if (fault != NoFault)
             return fault;
 
+        // look if the physical address is uncacheable
+        // and set the flag accordingly
+        Addr paddr = req->getPaddr();
+        if(pma->isUncacheable(paddr, req->getSize())){
+            req->setFlags(Request::UNCACHEABLE);
+            DPRINTF(TLB, "Access to address %#x turned to uncacheable.", paddr);
+        }
+
```

Comment:
I found there some common code section to add the `Request::UNCACHEABLE` flags to Request Packet. 

I think we can have the abstract method `checkSE` in `BasePMAChecker` and implement it in the `PMAChecker::checkSE` like

```cpp
void
PMAChecker::checkSE(const RequestPtr &req)
{
    if (isUncacheable(req->getPaddr(), req->getSize())) {
            req->setFlags(Request::UNCACHEABLE);
            DPRINTF(TLB, "Access to address %#x turned to uncacheable.", req->getPaddr());
    }
}
```
It won't require to change the `BasePMAChecker` to `PMAChecker`. How do you think?

Reply:
@iliasslasri,

Please reference the pervious PR https://github.com/gem5/gem5/pull/846, I feel that this PR revert the BasePMAChecker changes.